tests/test_correct_fsr_volume/test_correct_fsr_volume.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+
+import os
+import sys
+sys.path.insert(0, os.pardir)
+sys.path.insert(0, os.path.join(os.pardir, 'openmoc'))
+from testing_harness import TestHarness
+from input_set import PinCellInput
+
+
+class CorrectFSRVolumeTestHarness(TestHarness):
+    """Tests Geometry::correctFSRVolume() method."""
+
+    def __init__(self):
+        super(CorrectFSRVolumeTestHarness, self).__init__()
+        self.input_set = PinCellInput()
+        self._result = ''
+
+    def _generate_tracks(self):
+        """Generate Tracks and segments and correct the FSR volume for
+        the fuel."""
+
+        # Always use 1 thread for FSR reproducibility
+        self.track_generator.setNumThreads(1)
+        self.track_generator.generateTracks()
+
+        # Correct FSR volumes with the appropriate number of threads
+        self.track_generator.setNumThreads(self.num_threads)
+        old_volume = self.track_generator.getFSRVolume(1)
+        self.track_generator.correctFSRVolume(1, 2.7)
+        new_volume = self.track_generator.getFSRVolume(1)
+
+        # Save results for one thread
+        self._result += '{0: 1.6f}'.format(old_volume)
+        self._result += '{0: 1.1f}'.format(new_volume)
```

Comment:
Should we report more decimal places for the new volume? I feel like we should since we want to ensure that the new volume - computed from the adjusted segment lengths - accurately matches the value assigned by the user to more than a single decimal place.